### DISCUSSION TITLE
Native BigInt PK returns string instead of number

### DISCUSSION THREAD

**Discussion by sonicgamer0721** [mikro-orm/mikro-orm#5839] [General]
### Describe the bug

If I use entity:
```
@PrimaryKey({ type: new BigIntType('number') })
id: number;
```
em.find() returns {id: '33420'}, not {id: 33420}, so I have to modify every entity like this:
```
export class NativeBigIntType extends BigIntType {
  constructor() {
    super('number');
  }
  toJSON(value: any): any {
    return +value;
  }
}
...
  @PrimaryKey({ type: NativeBigIntType })
  id: number;
```
Could you please fix this?
Many thanks!

### Reproduction

No reproduction. Maybe the description is clear enough?

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.0

### Node.js version

20.15.0

### Operating system

Debian

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
> No reproduction. Maybe the description is clear enough?

Without actual reproduction, it's hard to say what is happening. There are tests that are working fine, I can't fix something I don't see.

Based on what you say, it sounds like you only observe this on the JSON, so not on your entity, but only if you serialize it?

**Reply by sonicgamer0721** [reply]
Yes, you are right!!! I test it again and find that when return data to client(browser) as json, the value of id changes to a string, and in server it is a number!
Is this a bug?

**Reply by B4nan** [reply]
Looking at the type implementation, its clear where this is coming from:

https://github.com/mikro-orm/mikro-orm/blob/e925d661383bc9fc686497f56e8fc3f519059af7/packages/core/src/types/BigIntType.ts#L40

I would rather not change this, as it feels potentially breaking and is easy to work around (just like you did).

**Reply by sonicgamer0721** [reply]
OK, got it. Thanks for your help!

**Reply by B4nan** [reply]
After a bit of thinking, I guess this should be fine to fix now, since it would only affect users who want to map to `number` anyway. Will see if I change my mind again over weekend or not :D

**Reply by B4nan** [reply]
FYI the next version will serialize to `number` here, let's see if someone complains, worst case I'll revert it.

ee24f1f0f017d8e7b4544f6cb1a39eab29dfb413

**Reply by KenjiChao** [reply]
@B4nan the same issue happens when I set the type to `integer`, could you also fix that?

**Reply by B4nan** [reply]
Not sure what you mean, this issue is about bigints, not integers.
